src/main/java/fredbot/FredBot.java

```diff
@@ -0,0 +1,301 @@
+    private static final String MESSAGE_LIST_HEADER = "Mr Busy over here has these tasks in his list:";
+    private static final String MESSAGE_MARK = "Ok and do you want a medal for that? I've marked this as done:";
+    private static final String MESSAGE_UNMARK = "Why am I not surprised... I've marked this task as not done yet:";
+    private static final String MESSAGE_WELCOME = "Yo I'm FredBot. What do you want with me?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final String FILE_PATH = "data/fredbot.txt";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        loadFredBot();
+        showWelcomeMessage();
+        while (true) {
+            String userInput = readUserInput();
+            executeCommand(userInput);
+            saveFredBot();
+        }
+    }
+
+    private static void loadFredBot() {
+        initTaskList();
+        File f = new File(FILE_PATH);
+        try {
+            loadTaskList(f);
+        } catch (FileNotFoundException e) {
+            initSaveFile(f);
+        }
+    }
+
+    private static void loadTaskList(File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String task = s.nextLine();
+            try {
+                processLine(task);
+            } catch (EmptyDescriptionException e) {
+                System.out.println(MESSAGE_EMPTY_DESCRIPTION);
+            }
+        }
+    }
+
+    private static void processLine(String task) throws EmptyDescriptionException {
+        String taskType = task.substring(0, 1);
+        String[] taskArgs = splitArgs(task);
+        switch (taskType) {
+        case "T":
+            allTasks.add(new Todo(taskArgs[2].trim()));
+            markDone(allTasks.get(count), taskArgs[1]);
+            count++;
+            break;
+        case "D":
+            allTasks.add(new Deadline(taskArgs[2].trim(), taskArgs[3].trim()));
+            markDone(allTasks.get(count), taskArgs[1]);
+            count++;
+            break;
+        case "E":
+            allTasks.add(new Event(taskArgs[2].trim(), taskArgs[3].trim(), taskArgs[4].trim()));
+            markDone(allTasks.get(count), taskArgs[1]);
+            count++;
+            break;
+        }
+    }
+
+    private static void markDone(Task t, String status) {
+        if (status.trim().equals("1")) {
+            t.markAsDone();
+        }
+    }
+
+    private static String[] splitArgs(String task) {
+        return task.split("\\|", 5);
+    }
+
+    private static void initSaveFile(File f) {
+        try {
+            new File("data").mkdir();
+            f.createNewFile();
+        } catch (IOException e) {
+            System.out.println(MESSAGE_ERROR);
+        }
+    }
```

Comment:
Variable names can be named more descriptively.

For literals, you have done that correctly for `MESSAGE_ERROR`. Good job! Now extend that to all of the other magic literals.